### PR TITLE
make `wccom` commands text translatable, add application-password section URL

### PR DESCRIPTION
**Pull request by rcstr**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Continues #34073.

### How to test the changes in this Pull Request:

1. Follow instructions from #34073
2. Run `wp wc com connect --force` to force URL being displayed
3. Confirms command works as expected
Screen Shot 2022-08-17 at 13 44 53

### CODE REVIEW

File: plugins/woocommerce/includes/cli/class-wc-cli-com-command.php [hunk 1]
```diff
@@ -16,6 +16,8 @@
  * @package WooCommerce
  */
 class WC_CLI_COM_Command {
+	const APPLICATION_PASSWORD_SECTION_URL = 'https://woocommerce.com/.com/my-account/#application-passwords';
```
Comment (jorgeatorres) [inline plugins/woocommerce/includes/cli/class-wc-cli-com-command.php]: This URL gives me a 404. Is it incorrect or just hasn't been enabled yet?

File: plugins/woocommerce/includes/cli/class-wc-cli-com-command.php [hunk 2]
```diff
@@ -152,20 +154,19 @@ public static function connect( array $args, array $assoc_args ) {
...
 
 				return;
 			}
 		}
 
-		// phpcs:ignore
-		// @todo add URL to application password section
 		if ( empty( $password ) ) {
-			$password = self::ask( 'Connection password:' );
+			WP_CLI::log( sprintf( __( 'If you don\'t have an application password (not your account password), generate a password from %s', 'woocommerce' ), esc_url( self::APPLICATION_PASSWORD_SECTION_URL ) ) );
```
Comment (jorgeatorres) [inline plugins/woocommerce/includes/cli/class-wc-cli-com-command.php:165]: Missing a `// translators:` comment.

```suggestion
			// translators: %s is a URL.
			WP_CLI::log( sprintf( __( 'If you don\'t have an application password (not your account password), generate a password from %s', 'woocommerce' ), esc_url( self::APPLICATION_PASSWORD_SECTION_URL ) ) );
```

File: plugins/woocommerce/includes/cli/class-wc-cli-com-command.php [hunk 3]
```diff
@@ -152,20 +154,19 @@ public static function connect( array $args, array $assoc_args ) {
...
+			$password = self::ask( __( 'Connection password:', 'woocommerce' ) );
 		}
 		$password = sanitize_text_field( $password );
 		if ( empty( $password ) ) {
-			WP_CLI::error( 'Invalid password.' );
+			WP_CLI::error( sprintf( __( 'Invalid password. Generate a new one from %s.', esc_url( self::APPLICATION_PASSWORD_SECTION_URL ) ), 'woocommerce' ) );
```
Comment (jorgeatorres) [inline plugins/woocommerce/includes/cli/class-wc-cli-com-command.php]: This is missing a `// translators:` comment above, but also the translation domain arg for `__()` and the URL (for `sprintf()`) have their places switched.

```suggestion
			// translators: %s is a URL.
			WP_CLI::error( sprintf( __( 'Invalid password. Generate a new one from %s.', 'woocommerce' ), esc_url( self::APPLICATION_PASSWORD_SECTION_URL ) ) );
```

**Review by jorgeatorres — CHANGES_REQUESTED** [review]
Hey @rcstr!

Overall, this looks good to me. Just a few tiny changes and we're good to go. Thanks!

File: plugins/woocommerce/includes/cli/class-wc-cli-com-command.php (hunk 1 above)
Comment (rcstr) [reply plugins/woocommerce/includes/cli/class-wc-cli-com-command.php]: fixed in 352cc7e7d51916a61e481d6682dc46f3d00af4a5

**Review by jorgeatorres — APPROVED** [review]
LGTM now. Thank you @rcstr!

**Pull request merged**